src/app/stack/stack-analyses.service.ts

```diff
@@ -63,6 +63,53 @@ export class StackAnalysesService {
     return null;
   }
 
+  getTokenStatus(url: string, uuid: string, params?: any) {
+    let getURL = url.concat('user/', uuid);
+    if (params) {
+      if (params['access_token']) {
+        let headers: Headers = new Headers();
+        headers.append('Authorization', 'Bearer ' + params['access_token']);
+        if (params['devcluster']) {
+          headers.append('x-3scale-account-secret', 'not-set');
+        }
+        return this.http.get(getURL, {
+          headers: headers,
+          params: {
+            'user_key': params['user_key']
+          }
+        })
+          .map(this.extractTokenData)
+          .toPromise()
+      }
+    }
+    return null;
+  }
+
+  putToken(url: string, uuid: string, token: string, params?: any) {
```

Comment:
Can this function be verbose? something like `linkSynkTokenWithUserID`?

Reply:
Updated it